### PR TITLE
Rework Java searchers

### PR DESCRIPTION
**Pull request by gjvoosten**
- Pull up sortBy and make common search query props optional with a default.
- Replace obsolete method.
- Extract DataLoaderKey enums into separate classes.
- Fix cross-database compatibility.
- Add support for running PostgreSQL as Docker containers.
- Remove support for SQLite.
- Introduce query builder and factor out common search criteria.
- Clean up use of spaces and caps in queries.

Part of NCI-Agency/anet#1705

### CODE REVIEW

File: src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java [hunk 1]
```diff
@@ -1,160 +1,77 @@
...
+  }
 
   @Override
-  public AnetBeanList<Task> runSearch(TaskSearchQuery query) {
-    final List<String> whereClauses = new LinkedList<String>();
-    final Map<String, Object> args = new HashMap<String, Object>();
-    final StringBuilder sql = new StringBuilder("/* MssqlTaskSearch */ SELECT tasks.*");
+  protected void buildQuery(TaskSearchQuery query) {
+    super.buildQuery(query);
+    qb.addSelectClause("COUNT(*) OVER() AS totalCount");
```
Comment (maradragan) [inline src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java]: Can't we do the totalcount only in one place, for all mssql queries? Or add an addTotalCount method we could call here?

File: src/main/java/mil/dds/anet/search/sqlite/SqliteAuthorizationGroupSearcher.java [hunk 2]
```diff
@@ -1,91 +1,23 @@
...
-
-    sql.append(" LIMIT :limit OFFSET :offset");
-
-    final List<AuthorizationGroup> list = getDbHandle().createQuery(sql.toString()).bindMap(sqlArgs)
-        .bind("offset", query.getPageSize() * query.getPageNum()).bind("limit", query.getPageSize())
-        .map(new AuthorizationGroupMapper()).list();
-
-    result.setList(list);
-    // Sqlite cannot do true total counts, so this is a crutch.
-    result.setTotalCount(result.getList().size());
```
Comment (maradragan) [inline src/main/java/mil/dds/anet/search/sqlite/SqliteAuthorizationGroupSearcher.java]: I think we are missing the totalcount in Sqlite queries now.

Comment (gjvoosten) [reply src/main/java/mil/dds/anet/search/sqlite/SqliteAuthorizationGroupSearcher.java]: That is now done in the `AnetBeanList` constructor.

File: src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java (hunk 1 above)
Comment (gjvoosten) [reply src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java]: In some cases the count needs ot be done in the outer query, so we can't always add it to the inner query. But a generic `addTotalCount(qb)` method seems like a good idea.

File: src/main/java/mil/dds/anet/search/sqlite/SqliteAuthorizationGroupSearcher.java (hunk 2 above)
Comment (gjvoosten) [reply src/main/java/mil/dds/anet/search/sqlite/SqliteAuthorizationGroupSearcher.java]: SQLite support has now been dropped in favour of PostgreSQL, which can do counts the same way as MS SQL Server.

**Pull request merged**